File: AdminPanel1/AdminPanel1.Web/Modules/ServerTypes/SouqCom/CategoryForm.ts
import { StringEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface CategoryForm {
    Name: StringEditor;
    Photo: StringEditor;
    Description: StringEditor;
}

export class CategoryForm extends PrefixedContext {
    static readonly formKey = 'SouqCom.Category';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!CategoryForm.init)  {
            CategoryForm.init = true;

            var w0 = StringEditor;

            initFormType(CategoryForm, [
                'Name', w0,
                'Photo', w0,
                'Description', w0
            ]);
        }
    }
}

File: AdminPanel1/AdminPanel1.Web/Modules/SouqCom/Category/CategoryForm.cs
using Serenity.ComponentModel;

namespace AdminPanel1.SouqCom.Forms;

[FormScript("SouqCom.Category")]
[BasedOnRow(typeof(CategoryRow), CheckNames = true)]
public class CategoryForm
{
    public string Name { get; set; }
    public string Photo { get; set; }
    public string Description { get; set; }
}